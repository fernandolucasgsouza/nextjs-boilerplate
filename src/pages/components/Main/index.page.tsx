import * as Styles from './styles.page';

const Main = () => (
    <Styles.Wrapper>
        <Styles.Header>
            <Styles.Logo
                src="/images/logo.png"
                alt="imagem de um Ã¡tomo que representa linguagem"
            ></Styles.Logo>
            <Styles.Title>Fernando Developer</Styles.Title>
        </Styles.Header>
        <Styles.Description>
            NextJs, TypeScript, e Styled Components.
        </Styles.Description>
        <Styles.Ilustration
            src="/images/programmer-rafiki.svg"
            alt="imagem de desenvolvedor de frente para um computador"
        />
    </Styles.Wrapper>
);

export default Main;
