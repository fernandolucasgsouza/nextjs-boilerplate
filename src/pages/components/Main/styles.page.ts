import styled from 'styled-components';

export const Wrapper = styled.main`
    background-color: #06092b;
    color: #fff;
    width: 100%;
    padding: 5rem;
    display: flex;
    flex-direction: column;
    text-align: center;
    justify-content: center;
`;
export const Header = styled.div`
    display: flex;
    justify-content: center;
    text-align: center;
    margin-bottom: 2rem;
`;
export const Logo = styled.img`
    width: 4rem;
`;

export const Title = styled.h1`
    font-size: 2.5rem;
`;

export const Description = styled.h2`
    font-size: 1.5rem;
    font-weight: 400;
`;

export const Ilustration = styled.img`
    width: min(30rem, 100%);
    margin: 3rem auto;
`;
