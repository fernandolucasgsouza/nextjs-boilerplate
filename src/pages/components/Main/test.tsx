import { render, screen } from '@testing-library/react';

import Main from './index.page';

describe('should render component <Main />', () => {
    it('should render the heading', () => {
        const { container } = render(<Main />);

        expect(
            screen.getByRole('heading', { name: /Fernando Developer/i })
        ).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });
});
