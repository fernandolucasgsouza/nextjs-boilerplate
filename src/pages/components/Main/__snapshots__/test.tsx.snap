// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component <Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  padding: 5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  margin-bottom: 2rem;
}

.c2 {
  width: 4rem;
}

.c3 {
  font-size: 2.5rem;
}

.c4 {
  font-size: 1.5rem;
  font-weight: 400;
}

.c5 {
  width: min(30rem,100%);
  margin: 3rem auto;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="imagem de um Ã¡tomo que representa linguagem"
      class="c2"
      src="/images/logo.png"
    />
    <h1
      class="c3"
    >
      Fernando Developer
    </h1>
  </div>
  <h2
    class="c4"
  >
    NextJs, TypeScript, e Styled Components.
  </h2>
  <img
    alt="imagem de desenvolvedor de frente para um computador"
    class="c5"
    src="/images/programmer-rafiki.svg"
  />
</main>
`;
